impl/src/main/java/org/jboss/weld/util/Observers.java

```diff
@@ -84,7 +85,8 @@ private Observers() {
 
     public static boolean isContainerLifecycleObserverMethod(ObserverMethod<?> method) {
         return CONTAINER_LIFECYCLE_EVENT_TYPES.contains(Reflections.getRawType(method.getObservedType()))
-                || (Object.class.equals(method.getObservedType()) && method instanceof ContainerLifecycleEventObserverMethod);
+                || (Object.class.equals(method.getObservedType()) && method.getObservedQualifiers().size() == 1
```

Comment:
@mkouba I am aware that the above behaviour is per spec, however I am not confused if we want to support this in a non-portable fashion or not? The earlier conversation around this issue gave me the impression that we do?